mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -94,33 +94,32 @@ def run(self, camera_data: Dict[str, CameraData], position_data: PositionData, o
         
         # Run one yolo inference on the provided frame at camera_data.frame
         center_camera_data = camera_data.get("center")
-        self.storageArray.append(center_camera_data.results[0])
-        
-        if(len(self.storageArray) != 0):
-            if(self.debug):
-                self.log("Interim processing")
-                self.log(self.storageArray)
-            returnVal = filter(self.storageArray)
-            records = self.analyze(returnVal)
-
-            self.colors, self.pattern = self.results(records)
-            if(self.debug):
-                self.log("self.logging results")
-                self.log(self.colors[0] + self.colors[1] + self.colors[2])
-                self.log("self.logging dict outside")
-                self.log(records)
+        if center_camera_data is not None:
+            # Something is detected
+            self.storageArray.append(center_camera_data.results[0].names)
```

Comment:
This isnt necessarily the best way of grabbing data. If you have multiple results you don't want to just grab the first one because it may be wrong, use some sort of confidence threshold. (Like [here](https://github.com/InspirationRobotics/RX24-perception/blob/e622c28b6ba2c97e9b99ef42c85cbe3e781b6331/camera/camera_core/camera.py#L195))

Reply:
Also .names returns a dictionary list of all the names. Please look closely at the linked code and understand how I got the name per bounding box.

Reply:
Also I believe the way you are trying to get the name of the detection is still wrong. Names returns a dictionary of all the names within the results object. Here are the [docs](https://arc.net/l/quote/aposyukn)